From 3544c8b35a7b6ce09cc959159cd83c51fd30f00a Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 29 Apr 2021 15:16:21 +0000
Subject: [PATCH] CMakeLists.txt: use sdbus-c++ from native build

* after passing CMAKE_TOOLCHAIN_FILE in CMAKE_ARGS we can build sdbus-c++
  but that only helps to build sdbus-c++ for target, for cross compilation
  just depend on sdbus-c++-tools-native and use that

* the target build is still needed, because then anbox is linked with -lsdbus-c++

| FAILED: src/anbox
| : && /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot-native/usr/bin/x86_64-webos-linux/x86_64-webos-linux-g++ -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse
 --sysroot=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/OE/build/luneos-honister/webos-ports/tm
p-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0=/usr/src/debug/anbox/3.0+gitAUTOINC+9de4e87cdd-r0                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/
3.0+gitAUTOINC+9de4e87cdd-r0=/usr/src/debug/anbox/3.0+gitAUTOINC+9de4e87cdd-r0                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysr
oot=                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden  -m64 -march=core2 -mtune=core2 -
msse3 -mfpmath=sse  --sysroot=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot -fvisibility=default -std=c++17 -pedantic -Wno-error=pedantic -Wall -Wextra -Wcast-al
ign -Wcast-qual -Wformat -Wredundant-decls -Wswitch-default -Wnon-virtual-dtor -Wold-style-cast -fno-permissive -DWAYLAND_SUPPORT -DEGL_NO_X11 -DUSE_PROTOBUF_CALLBACK_HEADER -std=c++17 -fPIC -DBINDERFS_PATH="\"/dev/binderfs\"" -g  -m64 -ma
rch=core2 -mtune=core2 -msse3 -mfpmath=sse  --sysroot=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-m
ap=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0=/usr/src/debug/anbox/3.0+gitAUTOINC+9de4e87cdd-r0                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports
/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0=/usr/src/debug/anbox/3.0+gitAUTOINC+9de4e87cdd-r0                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anb
ox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot=                      -fdebug-prefix-map=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot-native=  -fvisibility-inlin
es-hidden  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse  --sysroot=/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-need
ed -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed   -rdynamic external/backward-cpp/CMakeFiles/backward_object.dir/backward.cpp.o src/CMakeFiles/anbox.dir/main.cpp.o -o src/anbox -L/OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-6
4-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/build/src/../external/sdbus-cpp/src/sdbus-cpp-build -Wl,-rpath,"\$ORIGIN/../external/sdbus-cpp/src/sdbus-cpp-build:"  src/libanbox-core.a  -ldw  /OE/build/luneos-honister/webos-ports/tmp-gli
bc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_log.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/
lib/libboost_atomic.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_chrono.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/w
ork/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_regex.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/li
b/libboost_serialization.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_thread.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-gl
ibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_program_options.so.1.76.0  -lSDL2  -lSDL2  -lSDL2_image  -lSDL2_image  -lSDL2  -lSDL2  -lSDL2_image  -lSDL2_image  -llxc  -lutil  -llxc  -luti
l  -lsystemd  -lsystemd  external/cpu_features/libcpu_features.a  external/process-cpp-minimal/src/libprocess-cpp.a  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroo
t/usr/lib/libboost_iostreams.so.1.76.0  external/android-emugl/host/libs/librenderControl_dec.a  external/android-emugl/host/libs/libOpenGLESDispatch.a  external/android-emugl/shared/emugl/common/libemugl_common.a  -lpthread  -ldl  externa
l/android-emugl/host/libs/libGLESv1_dec.a  external/android-emugl/host/libs/libGLESv2_dec.a  external/android-emugl/shared/OpenglCodecCommon/libOpenglCodecCommon.a  src/libanbox-protobuf.a  -lprotobuf-lite  -lpthread  -lsdbus-c++  external
/xdg/libxdg.a  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_filesystem.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64
-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_system.so.1.76.0  /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot/usr/lib/libboost_
unit_test_framework.so.1.76.0 && :
| /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/11.0.1/ld: error: canno
t find -lsdbus-c++
| /OE/build/luneos-honister/webos-ports/tmp-glibc/work/core2-64-webos-linux/anbox/3.0+gitAUTOINC+9de4e87cdd-r0/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/11.0.1/ld: internal error in format_file_lineno, at ../../gold/dwarf_reader.cc:2278
| collect2: error: ld returned 1 exit status
| ninja: build stopped: subcommand failed.
| WARNING: exit code 1 from a shell command.

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a2abfdc..2c5043e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -59,7 +59,7 @@ add_library(anbox-protobuf
 target_link_libraries(anbox-protobuf
     ${PROTOBUF_LITE_LIBRARIES})
 
-set(XML2CPP ${CMAKE_BINARY_DIR}/external/sdbus-cpp/src/sdbus-cpp-build/tools/sdbus-c++-xml2cpp)
+set(XML2CPP sdbus-c++-xml2cpp)
 
 macro(DBusServer BaseName)
     add_custom_command(
